pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -654,17 +654,18 @@ private static void appendTime(StringBuilder sb, int hours, int minutes, int sec
     sb.append(':');
     sb.append(NUMBERS[seconds]);
 
-    // Add nanoseconds.
+    // Add microseconds, rounded.
     // This won't work for server versions < 7.2 which only want
     // a two digit fractional second, but we don't need to support 7.1
     // anymore and getting the version number here is difficult.
     //
-    if (nanos == 0) {
+    int microseconds = (int) Math.round(nanos / 1000.0);
```

Comment:
It looks like a precision loss is possible here.
I think `nanos/1000 + (nanos%1000+500/1000)` is better (or just `if (nanos%1000>500) micros++;`)